x/axelarnet/message_handler.go

```diff
@@ -129,17 +190,38 @@ func OnRecvMessage(ctx sdk.Context, k keeper.Keeper, ibcK keeper.IBCKeeper, n ty
 		Address: data.GetSender(),
 	}
 
+	// generate message id in advance to link fee payment and message
+	msgID := n.GenerateMessageID(ctx)
+
+	// deduct fee payment from packet if applicable
+	if msg.Fee != nil {
```

Comment:
Perhaps the `pay` method can be defined outside `Fee` type to also handle this case to reduce duplication